chain/chain/src/chain.rs

```diff
@@ -3524,8 +3524,12 @@ impl Chain {
         }
         let epoch_id = block_header.epoch_id();
         // Use epoch manager because block is not in DB yet.
+        // Let "epoch" be the epoch at which we enable stateless validation.
+        // We need to start producing the state witness from the last block of the previous epoch.
+        // To return true for the last block as input to this function, we need "next_protocol_version" to be
+        // StatelessValidationV0, hence we use block_header.hash() instead of block_header.prev_hash().
         let next_epoch_id =
-            self.epoch_manager.get_next_epoch_id_from_prev_block(block_header.prev_hash())?;
+            self.epoch_manager.get_next_epoch_id_from_prev_block(block_header.hash())?;
```

Comment:
Sanity check - will it work if the last block doesn't have a chunk, so for main transition this will be `false`?